tests/runtime-trace-tests/tracerunner.ts

```diff
@@ -0,0 +1,309 @@
+    tsAndPyFiles = tsFiles.concat(pyFiles)
+        .filter(f => !isDisabled(f))
+
+    tsAndPyFiles.forEach(file => {
+        it("should preserve the runtime semantics of " + path.basename(file), async function () {
+            this.timeout(10000)
+            await testTsOrPy(file)
+            return
+        });
+    });
+});
+
+async function testTsOrPy(tsOrPyFile: string): Promise<void> {
+    let ext = path.extname(tsOrPyFile)
+    let isPy = ext === ".py"
+    let isTs = ext === ".ts"
+    if (!isPy && !isTs)
+        return Promise.reject(new Error("Invald non-.py, non-.ts file: " + tsOrPyFile))
+
+    let baselineFile: string;
+    let baseline: string;
+    let recordBaseline = (bl: string) => {
+        baseline = bl
+        baselineFile = tsOrPyFile + ".baseline"
+        fs.writeFileSync(baselineFile, baseline)
+    }
+    if (isPy) {
+        let pyFile = tsOrPyFile
+        recordBaseline(await PY(pyFile))
+        // TODO(dazuniga): py2ts doesn't work in unit tests yet, once it does enable it here
+        // let tsFile = await testPy2Ts(pyFile)
+        // await testSts(tsFile)
+        // let pyFile2 = await testTs2Py(tsFile)
+    } else {
+        let tsFile = tsOrPyFile
+        recordBaseline(await TS(tsFile))
+        await testSts(tsFile)
+        let pyFile = await testTs2Py(tsFile)
+        // TODO(dazuniga): py2ts doesn't work in unit tests yet, once it does enable it here
+        // let tsfile2 = await testPy2Ts(pyFile)
+        // await testSts(tsfile2)
+    }
+
+    return;
+    // DSL for the tests
+    type Trace = string
+    type File = string
+    function STS(tsFile: string): Promise<Trace> { return compileAndRunStsAsync(tsFile) }
+    function TS(tsFile: string): Promise<Trace> { return compileAndRunTs(tsFile) }
+    function PY(pyFile: string): Promise<Trace> { return runPyAsync(pyFile) }
+    function TS2PY(tsFile: string): Promise<File> { return convertTs2Py(tsFile) }
+    function PY2TS(pyFile: string): Promise<File> { return convertPy2Ts(pyFile) }
+    // Test functions & helpers
+    async function testPy2Ts(pyFile: string): Promise<string> {
+        return testConversion(pyFile, true)
+    }
+    async function testTs2Py(tsFile: string): Promise<string> {
+        return testConversion(tsFile, false)
+    }
+    async function testConversion(inFile: string, isPy: boolean): Promise<string> {
+        let convert = isPy ? PY2TS : TS2PY
+        let runConverted = isPy ? TS : PY
+        let fnName = isPy ? "py2ts" : "ts2py"
+        let errFile = inFile + `.${fnName}_error`;
+        return convert(inFile)
+            .error(r => {
+                fs.writeFileSync(errFile, JSON.stringify(r))
+                return `${fnName} failed to convert '${inFile}'. Error saved at:\n${errFile}\nError is:\n${r}\n`
+            })
+            .then(async outFile => {
+                let outTrace = await runConverted(outFile)
+                if (!util.compareBaselines(outTrace, baseline)) {
```

Comment:
I think that `compareBaselines` has some decompiler specific stuff it does to the input like stripping some XML attributes. Might need to have two functions.

Reply:
Good to know